### PR TITLE
Fix Xfstest

### PR DESCRIPTION
**Pull request by PraveenPenguin**
fix xfstest
```
1-enable xfstest to run fedora/centos/RHEL
2-enable local.config file to handle environment variable
3-Rectify README File
```

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
Job log
[joblog.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/827092/joblog.txt)

File: fs/xfstests.py [hunk 1]
```diff
@@ -39,14 +39,27 @@ def setUp(self):
...
+        detected_distro = distro.detect()
+
+        packages = ['e2fsprogs', 'automake', 'gcc', 'quota', 'attr',
+                    'make',  'xfsprogs',  'gawk', 'fio', 'dbench']
+
+        if 'Ubuntu' in detected_distro.name:
+            packages.extend(['xfslibs-dev', 'uuid-dev', 'libtool-bin', 'libuuid1',
+                             'libattr1-dev', 'libacl1-dev', 'libgdbm-dev', 'uuid-runtime'])
+
+        elif detected_distro.name in ['centos', 'fedora', 'redhat']:
```
Comment (narasimhan-v) [inline fs/xfstests.py]: Are we not considering suse here ?

Comment (PraveenPenguin) [inline fs/xfstests.py]: @narasimhan-v no

File: fs/xfstests.py.data/local.config [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export TEST_DEV=/dev/loop0
+export TEST_DIR=/mnt/test
+export SCRATCH_DEV=/dev/loop1
+export SCRATCH_MNT=/mnt/scratch
```
Comment (narasimhan-v) [inline fs/xfstests.py.data/local.config]: Does the user need to edit this file frequently ?
If yes, can it be handled with yaml ?

Comment (PraveenPenguin) [inline fs/xfstests.py.data/local.config]: yes it need to edit via user as it require valid input (Device) of test trigger , i think both way is fine as we need to have in  config  so i think editing in yaml than dump those entry config seems not good so it is better to edit config file directly  i think this approach looks me ok as xfstest follow this  ,

File: fs/xfstests.py [hunk 3]
```diff
@@ -39,14 +39,27 @@ def setUp(self):
...
+                             'libaio-devel', 'libuuid-devel', 'openssl-devel', 'xfsprogs-devel',
+                             'btrfs-progs-devel'])
+        else:
+            self.skip("test not supported in %s" % detected_distro.name)
+
         for package in packages:
             if not sm.check_installed(package) and not sm.install(package):
-                self.error("Fail to install %s required for this test." %
-                           package)
+                self.skip("Fail to install %s required for this test." %
```
Comment (balamuruhans) [inline fs/xfstests.py]: shouldn't it be error, as we skip tests if package is not there or if env is unsupported, but when test tries to install some package and if it fails to install its better to error out.

File: fs/xfstests.py.data/local.config (hunk 2 above)
Comment (narasimhan-v) [inline fs/xfstests.py.data/local.config]: @PraveenPenguin Okay, for now we can have this minimalistic version, with `TODO` added indicating creating FS, mounting the disk/loop devices, etc

File: fs/xfstests.py (hunk 3 above)
Comment (PraveenPenguin) [inline fs/xfstests.py]: i think should be ok as if these package not installed as these package is not available so not seems me test error  , test is not able to run so i think we can skip the test and in report it kind of skip what you say ?

File: fs/xfstests.py.data/local.config (hunk 2 above)
Comment (PraveenPenguin) [inline fs/xfstests.py.data/local.config]: @narasimhan-v sure thanks will add in TODO

File: fs/xfstests.py (hunk 3 above)
Comment (balamuruhans) [inline fs/xfstests.py]: fine then, Looks good otherwise

Comment (PraveenPenguin) [inline fs/xfstests.py]: @balamuruhans thanks

**Comment by PraveenPenguin** [comment]
closing in favour of #314

**Pull request closed without merging**